.github/actions/coverage-gate/check-per-file-coverage.py

```
@@ -0,0 +1,118 @@
+                lines = cls.findall("lines/line")
+                if not lines:
+                    continue
+
+                valid = len(lines)
+                covered = sum(1 for l in lines if int(l.get("hits", 0)) > 0)
+
+                if fname not in file_stats:
+                    file_stats[fname] = {"valid": 0, "covered": 0}
+                file_stats[fname]["valid"] += valid
+                file_stats[fname]["covered"] += covered
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)